### PR TITLE
Filter values as a string not an object

### LINKED ISSUE #7389: Allow filtering of time list activities by metadata
**Is your feature request related to a problem? Please describe.**
In addition to the name, filtering on other metadata of an activity is beneficial

**Describe the solution you'd like**
Allow filtering activities on metadata in addition to just the 'name'.
Allow users to specify which specific properties they would like searched for an activity. In the absence of these specified properties, by default, search for values in `properties`.
```
activity: {
    name: "First",
    properties: {
           location: "Grand Hall"
    }
}
```

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #7389 

### Describe your changes:
Use the value of a property for filtering metadata as a string if it is not undefined.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [ ] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by scottbell** [comment]
I'm not sure how to test this given the instructions on the ticket. Is `Duration` metadata? If so, it doesn't appear to be functioning correctly. I tried entering as 15:
Screenshot 2024-02-01 at 3 58 17 PM

which rendered no items:
Screenshot 2024-02-01 at 3 58 27 PM

Putting in the exact duration didn't seem to work either:
Screenshot 2024-02-01 at 3 56 58 PM

One other confusing thing: In edit mode, the time conductor bounds seem to be ignored? It makes adding filters rather confusing:

https://github.com/nasa/openmct/assets/9853862/c8c19739-ca85-41ea-8c6e-7509a7c61725

**Review by scottbell — CHANGES_REQUESTED** [review]
Having trouble testing this. We should make the testing instructions more clear with images, or a video. Also, we should add an e2e test.

**Comment by shefalijoshi** [comment]
We have a unit test for this already. We can add e2e test later.

**Comment by shefalijoshi** [comment]
> I'm not sure how to test this given the instructions on the ticket. Is `Duration` metadata? If so, it doesn't appear to be functioning correctly. I tried entering as 15: Screenshot 2024-02-01 at 3 58 17 PM
> 
> which rendered no items: Screenshot 2024-02-01 at 3 58 27 PM
> 
> Putting in the exact duration didn't seem to work either: Screenshot 2024-02-01 at 3 56 58 PM
> 
> One other confusing thing: In edit mode, the time conductor bounds seem to be ignored? It makes adding filters rather confusing:
> Screen.Recording.2024-02-01.at.4.03.57.PM.mov

Duration can be metadata if you define it in the sourcemap or if it is part of an activity's property like this:
`activity: {properties: {duration: 15}}`

One other confusing thing: In edit mode, the time conductor bounds seem to be ignored? It makes adding filters rather confusing: This is working as designed.

We have a unit test for this already. We can add e2e test later.

I've added documentation for sourceMaps is here: fix-filter-metadata

File: src/plugins/plan/util.js [hunk 1]
```diff
@@ -62,9 +63,9 @@ export function getValidatedData(domainObject) {
           groupActivity.filterMetadataValues = [];
           sourceMap.filterMetadata.forEach((property) => {
             const value = _.get(activity, property);
-            groupActivity.filterMetadataValues.push({
-              value
-            });
+            if (value !== undefined) {
```
Comment (scottbell) [inline src/plugins/plan/util.js:66]: can this ever be null? should that be checked too?

**Comment by scottbell** [comment]
Filtering on activities and with commas appeared to work:
Screenshot 2024-02-02 at 1 41 38 PM

as does meta-filtering:
Screenshot 2024-02-02 at 1 41 47 PM
Screenshot 2024-02-02 at 1 42 48 PM

with both filters in use, are they supposed to be OR'd together? That looks like the current implementation:
Screenshot 2024-02-02 at 1 44 51 PM

Also, PR doesn't look like it addresses @charlesh88's other issues, in particular:
>We can't filter on values with "." in them - why not? If the user is trying to filter on a decimal number, they won't be able to.
The error messaging for the input is very poor and provides no concrete feedback of what the problem is or how to address it. This needs to be addressed from a text and layout standpoint.

Screenshot 2024-02-02 at 1 46 44 PM

**Comment by scottbell** [comment]
Looks like one can enter `.` now in filters too:
image

**Review by scottbell — APPROVED** [review]
LGTM

**Pull request merged**